src/bench/merkle_root.cpp

```diff
@@ -7,21 +7,34 @@
 #include <random.h>
 #include <uint256.h>
 
+#include <cassert>
 #include <vector>
 
 static void MerkleRoot(benchmark::Bench& bench)
 {
-    FastRandomContext rng(true);
-    std::vector<uint256> leaves;
-    leaves.resize(9001);
-    for (auto& item : leaves) {
+    FastRandomContext rng{/*fDeterministic=*/true};
+
+    std::vector<uint256> hashes{};
+    hashes.resize(9001);
+    for (auto& item : hashes) {
         item = rng.rand256();
     }
-    bench.batch(leaves.size()).unit("leaf").run([&] {
-        bool mutation = false;
-        uint256 hash = ComputeMerkleRoot(std::vector<uint256>(leaves), &mutation);
-        leaves[mutation] = hash;
-    });
+
+    constexpr uint256 expected_root{"d8d4dfd014a533bc3941b8663fa6e7f3a8707af124f713164d75b0c3179ecb08"};
+    for (bool mutate : {false, true}) {
+        bench.name(mutate ? "MerkleRootWithMutation" : "MerkleRoot").batch(hashes.size()).unit("leaf").run([&] {
+            std::vector<uint256> leaves;
+            leaves.reserve((hashes.size() + 1) & ~1ULL); // capacity rounded up to even
```

Comment:
This reminds me of the [Obfuscated C Code Contest](https://github.com/ioccc-src/winner/blob/master/2024/burton/prog.c). I consider the dumb version easier to read:

```cpp
x % 2 == 0 ? x : x + 1
```
Remember that other people could get confused easier than you.

Reply:
> maybe put it in an inline function RoundUpToEven(n)

I did that before, reviewers found it complicated.